core/Access.php

```diff
@@ -241,11 +246,12 @@ protected function reloadAccessSuperUser()
      */
     public function setSuperUserAccess($bool = true)
     {
+        $this->hasSuperUserAccess = (bool) $bool;
+
         if ($bool) {
-            $this->reloadAccessSuperUser();
+            $this->makeSureLoginNameIsSet();
```

Comment:
Don't think it's needed. It would rather end up in loading more sites eg if `superuser` sites were already set we do not need to reload them.